### PR TITLE
Generalize path to bash

### PR DESCRIPTION
**Pull request by brianyu28**
Requested by @crossroads1112 in order to support the Python version of help50-client.

### CODE REVIEW

**Comment by crossroads1112** [comment]
This PR generalizes the path to `ls`, `cp`, which I think is a good thing, but only the generalization of the `bash` executable location is necessary for the  because `ptyprocess`

**Comment by cmlsharp** [comment]
This PR generalizes the path to `ls`, `cp`, which I think is a good thing, but only the generalization of the `bash` executable location is necessary for the `help50` port  because `ptyprocess`  (what `pexpect` uses on the backend) determines the full path of the executable you give it and runs that. E.g. when you do `PtyProcess.spawn(["bash"])`, bash gets `/usr/bin/bash` in `argv[0]` which affects the error message it prints. 

For reference, we need `PtyProcess` instead of e.g. `subprocess` because we don't want the process we're running to know that it is writing to a pipe since that causes many programs (i.e. clang) to disable colored output.

Either way, I think this is a good change since `help50 /usr/bin/ls adsgfdshg` and `help50 ls adsgfdshg` should result in the same hint.

**Comment by dmalan** [comment]
Hm, is there some way to parameterize `ptyprocess` not to put full paths in `argv`, as by passing it a `PATH` or such? Else it sounds like students will see just `bash` if they run a command but `/bin/bash` if they run that command through `help50`?

**Review by dmalan — COMMENTED** [review]
Hm, is there some way to parameterize `ptyprocess` not to put full paths in `argv`, as by passing it a `PATH` or such? Else it sounds like students will see just `bash` if they run a command but `/bin/bash` if they run that command through `help50`?

**Comment by cmlsharp** [comment]
Unfortunately, there doesn't seem to be, no. The [spawn](https://github.com/pexpect/ptyprocess/blob/master/ptyprocess/ptyprocess.py#L205) method, calls PytProcess's own internal [which](https://github.com/pexpect/ptyprocess/blob/master/ptyprocess/util.py#L64) function which always returns the full path.

File: regex.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FILE_PATH = "(?:(?:.*)\/(?:[^/]+)\/)?"
```
Comment (dmalan) [inline regex.py:1]: What's the `(?:.*)` meant to capture? Why not, e.g., below?

```
FILE_PATH = "(?:\/(?:[^/]+\/)*)?"
```

**Review by dmalan — APPROVED** [review]

**Pull request merged**